packages/oui-button/src/index.spec.js

```diff
@@ -0,0 +1,109 @@
+describe('ouiButton', () => {
+  let $componentController
+
+  beforeEach(angular.mock.module('oui.button'))
+
+  beforeEach(inject((_$componentController_) => {
+    $componentController = _$componentController_
+  }))
+
+  describe('Controller', () => {
+    it('should exist', () => {
+      const ctrl = $componentController('ouiButton', {
+        $attrs: {},
+        $element: {},
+        $log: {}
+      })
+
+      expect(ctrl).toBeDefined()
+    })
+  })
+
+  describe('Component', () => {
+    let $rootScope
+    let $compile
+
+    beforeEach(inject((_$rootScope_, _$compile_) => {
+      $rootScope = _$rootScope_
+      $compile = _$compile_
+    }))
+
+    it('should display a button with value of attribute text, and is type="button" by default', () => {
+      let element = angular.element('<oui-button text="foo"></oui-button>')
+      let scope = $rootScope.$new()
+
+      $compile(element)(scope)
+      scope.$digest()
+
+      let button = element.find('button').eq(0)
+      expect(button.text().trim()).toBe('foo')
+      expect(button.attr('type')).toBe('button')
+    })
+
+    it('should have an attribute id and name on the button, and removed on the root component', () => {
+      let element = angular.element('<oui-button id="foo" name="bar"></oui-button>')
+      let scope = $rootScope.$new()
+
+      $compile(element)(scope)
+      scope.$digest()
+
+      let button = element.find('button').eq(0)
+      expect(element.attr('id')).toBe(undefined)
+      expect(button.attr('id')).toBe('foo')
+
+      expect(element.attr('name')).toBe(undefined)
+      expect(button.attr('name')).toBe('bar')
+    })
+
+    it('should have an attribute aria-label on the button, and removed on the root component', () => {
+      let element = angular.element('<oui-button aria-label="foo"></oui-button>')
+      let scope = $rootScope.$new()
+
+      $compile(element)(scope)
+      scope.$digest()
+
+      expect(element.attr('aria-label')).toBe(undefined)
+      expect(element.find('button').eq(0).attr('aria-label')).toBe('foo')
+    })
+
+    it('should have a primary next step button', () => {
```

Comment:
You are testing the `next` state, but what about the `previous` state, somebody could break it and we will never know ?

Same thing about `secondary` and `link` states.

Reply:
I don't know. In this case, i'll have to test also `disabled="true"` and `disabled="false"` to be complete

Here, I prefer to test the attribute and its behaviour more than its value.
But why not if necessary :)